client/signup/jetpack-connect/plans-landing.jsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import page from 'page';
+import React from 'react';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import PlansGrid from './plans-grid';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { selectPlanInAdvance } from 'state/jetpack-connect/actions';
+import QueryPlans from 'components/data/query-plans';
+
+const CALYPSO_JETPACK_CONNECT = '/jetpack/connect';
+
+const PlansLanding = React.createClass( {
+
+	propTypes: {
+		sites: React.PropTypes.object,
+		sitePlans: React.PropTypes.object.isRequired,
+		intervalType: React.PropTypes.string
+	},
+
+	getDefaultProps() {
+		return {
+			intervalType: 'yearly',
+			siteSlug: '*'
+		};
+	},
+
+	componentDidMount() {
+		this.props.recordTracksEvent( 'calypso_jpc_plans_view' );
+	},
+
+	storeSelectedPlan( cartItem ) {
+		this.props.recordTracksEvent( 'calypso_jpc_plans_store_plan', {
+			plan: cartItem ? cartItem.product_slug : 'free'
+		} );
+		this.props.selectPlanInAdvance(	cartItem ? cartItem.product_slug : 'free', '*', );
+
+		setTimeout( () => {
+			page.redirect( CALYPSO_JETPACK_CONNECT );
+		}, 25 );
+	},
+
+	render() {
+		return (
+			<div>
+				<QueryPlans />
+				<PlansGrid { ...this.props } onSelect={ this.storeSelectedPlan } />
+			</div>
+		);
+	}
+} );
+
+export default connect(
+	() => {
+		return {
+			sitePlans: {},
+			calypsoStartedConnection: true
```

Comment:
well, for this page (anyone who have seen the plans landing page) we would want them to finish in their plan page once the flow is complete, so we don't care if they arrived from wp-admin